### PR TITLE
Adding Resume model

### LINKED ISSUE #26: Adding the ResuMe model
- Creating the ResuMe table
- Testing the ResuMe table

### PR DESCRIPTION
**Pull request by AlonShakaroff**
- Adding the Resume model
- Adding related tests

fix #26 
Please merge that PR only after merging #47

### CODE REVIEW

**Review by OmerCS8 — DISMISSED** [review]
LGTM

**Review by MatanP12 — CHANGES_REQUESTED** [review]
Please resolve the conflicts so I could pull the PR

**Review by OmerCS8 — APPROVED** [review]
LGTM

**Review by TomerNewmanPrograms — DISMISSED** [review]
Lgtm, just squash

**Review by MatanP12 — APPROVED** [review]
LGTM

**Review by MatanP12 — APPROVED** [review]
LGTM

**Review by OmerCS8 — DISMISSED** [review]
LGTM

**Review by TomerNewmanPrograms — DISMISSED** [review]
Lgtm

**Review by yulisuliman — DISMISSED** [review]
Lgtm

File: posts/tests.py [hunk 1]
```diff
@@ -1,3 +1,89 @@
...
+# Check that the Resume values are the same as the Resume inputs.
+def test_new_resume(new_resume):
+    assert new_resume.author.username == USERNAME
+    assert new_resume.description == DESCRIPTION
+    # assert len(new_resume.rating_set.all()) == 0  #add after adding rating
+
+
+# Check if the Resume can be referred as a post.
+@pytest.mark.django_db()
+def test_resume_as_post(new_resume):
```
Comment (natifridman) [inline posts/tests.py:39]: You can check if `new_resume` is an instance of a post:
```python
assert isinstance(new_resume, Post)
```

File: posts/tests.py [hunk 2]
```diff
@@ -1,3 +1,89 @@
...
+    resume = post.resume
+    assert new_resume == resume
+
+
+# Check if the Resume is saved in the database.
+@pytest.mark.django_db()
+def test_persist_resume(new_resume):
+    assert new_resume not in Resume.objects.all()
+    new_resume.author.save()
+    new_resume.save()
```
Comment (natifridman) [inline posts/tests.py:56]: Why do you need to save it twice?

File: posts/tests.py [hunk 3]
```diff
@@ -1,3 +1,89 @@
...
+    assert new_resume in Resume.objects.all()
+    assert user in User.objects.all()
+    user.delete()
+    assert new_resume not in Resume.objects.all()
+    assert user not in User.objects.all()
+
+
+# verify that it is impossible to create a Resume with invalid parameters.
+def test_resume_invalid_args():
```
Comment (natifridman) [inline posts/tests.py:87]: I think you can use parametrize here to test some invalid values cases

**Review by natifridman — CHANGES_REQUESTED** [review]
I think you can use [test classes](https://docs.pytest.org/en/7.1.x/getting-started.html#group-multiple-tests-in-a-class) here so you won't need to create a user every time

File: posts/tests.py (hunk 3 above)
Comment (AlonShakaroff) [reply posts/tests.py]: Ok, switched it to parametrize.

**Review by AlonShakaroff — COMMENTED** [review]
Addressed all the reviews by @natifridman

**Review by OmerCS8 — APPROVED** [review]
LGTM

**Review by MatanP12 — DISMISSED** [review]
LGTM

**Review by TomerNewmanPrograms — APPROVED** [review]
lgtm

**Review by yulisuliman — DISMISSED** [review]
Look good!

**Review by MatanP12 — DISMISSED** [review]
LGTM

**Review by OmerCS8 — DISMISSED** [review]
LGTM

**Review by TomerNewmanPrograms — DISMISSED** [review]

**Review by yulisuliman — APPROVED** [review]
Lgtm

**Review by OmerCS8 — DISMISSED** [review]
LGTM

**Review by MatanP12 — DISMISSED** [review]
LGTM

File: posts/tests.py [hunk 4]
```diff
@@ -1,3 +1,91 @@
...
+    # Check if the Resume can be referred as a post.
+    def test_resume_as_post(self, new_resume):
+        assert len(Post.objects.all()) == 0
+        new_resume.author.save()
+        new_resume.save()
+        assert isinstance(new_resume, Post)
+        assert len(Post.objects.all()) == 1
+        post = Post.objects.all().first()
+        assert hasattr(post, 'resume') is True
+        assert hasattr(post, 'poll') is False
```
Comment (abiton1) [inline posts/tests.py:58]:
```suggestion
        assert hasattr(post, 'resume')
        assert not hasattr(post, 'poll')
```
It is redundant to check if True or False

**Review by yulisuliman — DISMISSED** [review]

**Review by MatanP12 — APPROVED** [review]
LGTM

File: posts/tests.py (hunk 4 above)
Comment (AlonShakaroff) [reply posts/tests.py]: Ok, thank you, I changed that 👍 @abiton1

**Review by OmerCS8 — APPROVED** [review]
LGTM

**Review by TomerNewmanPrograms — APPROVED** [review]

**Review by natifridman — COMMENTED** [review]
Added some comments in the tests, but looks good!

File: posts/tests.py [hunk 5]
```diff
@@ -1,3 +1,91 @@
...
+        assert new_resume.author.username == USERNAME
+        assert new_resume.description == DESCRIPTION
+
+    # Check if the Resume can be referred as a post.
+    def test_resume_as_post(self, new_resume):
+        assert len(Post.objects.all()) == 0
+        new_resume.author.save()
+        new_resume.save()
+        assert isinstance(new_resume, Post)
+        assert len(Post.objects.all()) == 1
```
[the remark this answers is not in the document]
Comment (AlonShakaroff) [reply posts/tests.py:55]: Actually, it's probably redundant, I'll remove that.

File: posts/tests.py [hunk 6]
```diff
@@ -1,3 +1,91 @@
...
+        post = Post.objects.all().first()
+        assert hasattr(post, 'resume')
+        assert not hasattr(post, 'poll')
+        resume = post.resume
+        assert new_resume == resume
```
[the remark this answers is not in the document]
Comment (AlonShakaroff) [reply posts/tests.py:60]: > Why do you need to check this if it's an instance of a post? If you need it, I think you could do something like
> 
> ```python
> assert Post.objects.filter(resume= new_resume).exists()
> ```

To check if the inheritance works properly.
I'll change the test to the line you suggested, thank you.

**Comment by AlonShakaroff** [comment]
Fixed Nati's notes,
and also changed the test method name to more informative names.
@natifridman

File: posts/tests.py [hunk 7]
```diff
@@ -1,3 +1,87 @@
...
+        assert new_resume.description == DESCRIPTION
+
+    # Check if the Resume can be referred as a post.
+    def test_inheritance_of_resume_from_post(self, persist_resume):
+        post = Post.objects.all().first()
+        assert Post.objects.filter(resume=persist_resume).exists()
+        assert hasattr(post, 'resume')
+        assert not hasattr(post, 'poll')
+        resume = post.resume
+        assert persist_resume == resume
```
Comment (OmerCS8) [inline posts/tests.py:56]: Aren't lines 55,56 redundant after line 52?

**Review by OmerCS8 — CHANGES_REQUESTED** [review]

File: posts/tests.py (hunk 7 above)
Comment (AlonShakaroff) [reply posts/tests.py:56]: You're right, I'm fixing it right now.

**Review by OmerCS8 — APPROVED** [review]
LGTM

**Review by MatanP12 — DISMISSED** [review]
LGTM

**Review by TomerNewmanPrograms — APPROVED** [review]

**Review by natifridman — APPROVED** [review]

**Review by OmerCS8 — APPROVED** [review]
LGTM

**Review by TomerNewmanPrograms — APPROVED** [review]

**Review by yulisuliman — APPROVED** [review]

**Review by natifridman — APPROVED** [review]

**Pull request merged**